### PR TITLE
Update linting process

### PR DESCRIPTION
**Pull request by timfelle**
This PR includes changes to improve the linting process and adds a dummy change to observe its effect.
We now only stop merger of a PR if it actively degrades the quality of the repository.
Otherwise we just report what can be improved.

Closes issue #1423

### CODE REVIEW

**Comment by timfelle** [comment]
@njansson maybe its nice to squash this. Lots of kinda empty commits

**Comment by timofeymukha** [comment]
So essentially the higher the repo score will be, the higher the requirements for incoming code, right :P?

File: .github/workflows/check_lint.yml [hunk 1]
```diff
@@ -144,7 +170,7 @@ jobs:
                   grep -oP '(?<=\>\|)[^\|\<]+(?=\|\<)')
             printf ": $score\n"
 
-            if (($(echo "$score < $changed_min" | bc -l))); then
+            if (($(echo "$score < 10" | bc -l))); then
```
Comment (MartinKarp) [inline .github/workflows/check_lint.yml:173]: Is this for all new files or only if the overall score does not deteriorate (regardless if new files were added or not)?

**Review by MartinKarp — COMMENTED** [review]
This looks great! Just a question regarding how we handle adding new files.

**Comment by timfelle** [comment]
> So essentially the higher the repo score will be, the higher the requirements for incoming code, right :P?

Sort of yes. We only check if the entire repository was degraded, we don't check for every file.
That could also be a way to do it. Make sure that any individual file changes are better than the average.

File: .github/workflows/check_lint.yml (hunk 1 above)
Comment (timfelle) [reply .github/workflows/check_lint.yml:173]: We only exit with an error for the CI if the score of the entire repository goes down.

But we always print a list of possible improvements, which is the 10 there.

Comment (timfelle) [reply .github/workflows/check_lint.yml:173]: New files are not handled differently than old files here.

Comment (MartinKarp) [reply .github/workflows/check_lint.yml:173]: Top! Sounds great to me.

**Review by MartinKarp — APPROVED** [review]

**Comment by timofeymukha** [comment]
I guess there is some fairness in not demanding more than what we have now. But I feel like this is temporary in nature, since eventually we would push the score up, and so be in the same dilemma as now. I think in the end of the day, we need to decided what rules are really important to us and lint those hard. People seem to mostly be upset about spaces, for example. I would rather sacrifice some space rules, but be sure that all code is 80col and uses `only`. That said, I am not against merging this as.

**Comment by timfelle** [comment]
> I guess there is some fairness in not demanding more than what we have now. But I feel like this is temporary in nature, since eventually we would push the score up, and so be in the same dilemma as now. I think in the end of the day, we need to decided what rules are really important to us and lint those hard. People seem to mostly be upset about spaces, for example. I would rather sacrifice some space rules, but be sure that all code is 80col and uses `only`. That said, I am not against merging this.

Yea over lunch i also considered making two different versions of the linter rules. Exactly as you suggest with one being a hard requirement always, and the other to be more lax. But I am still of the mind that if we have a rule, it must be followed by all. I am not against removing rules that we feel do not make sense, but I am against not enforcing the rules we decide to have.

**Comment by njansson** [comment]
> @njansson maybe its nice to squash this. Lots of kinda empty commits

Sure, I'll put a squash commit

**Pull request merged**